packages/node/src/indexer/blockDispatcher/worker-block-dispatcher.service.ts

```diff
@@ -214,7 +219,7 @@ export class WorkerBlockDispatcherService
 
         const start = new Date();
         await memoryLock.acquire();
-        await worker.fetchBlock(height, blockSpecVersion);
+        await pendingBlock;
```

Comment:
Doesn't moving this call break the memory lock?

Reply:
I think locking while fetching blocks will limit the number of fetches to 1 at a time. instead, we can just lock while freeing up memory. Before fetching blocks we call `memoryLock.waitForUnlock()`. 

Also, we can change the `waitForWorkerBatchSize`  function to only lock if memory is unavailable, right now it locks everytime it is called.